build-system/tasks/lint.js

```diff
@@ -17,130 +17,121 @@
 
 const argv = require('minimist')(process.argv.slice(2));
 const config = require('../test-configs/config');
-const debounce = require('gulp-debounce');
-const eslint = require('gulp-eslint');
-const eslintIfFixed = require('gulp-eslint-if-fixed');
+const fs = require('fs');
 const globby = require('globby');
-const gulp = require('gulp');
-const lazypipe = require('lazypipe');
 const path = require('path');
-const watch = require('gulp-watch');
 const {
   log,
   logLocalDev,
   logOnSameLine,
   logOnSameLineLocalDev,
 } = require('../common/logging');
 const {cyan, green, red, yellow} = require('kleur/colors');
+const {ESLint} = require('eslint');
 const {getFilesChanged, getFilesFromArgv} = require('../common/utils');
 const {gitDiffNameOnlyMaster} = require('../common/git');
 const {maybeUpdatePackages} = require('./update-packages');
-const {watchDebounceDelay} = require('./helpers');
 
 const rootDir = path.dirname(path.dirname(__dirname));
 
+const options = {
+  fix: argv.fix,
+  reportUnusedDisableDirectives: 'error',
+};
+
 /**
- * Initializes the linter stream based on globs
- *
- * @param {!Object} globs
- * @param {!Object} streamOptions
- * @return {!ReadableStream}
+ * Runs the linter on the given set of files.
+ * @param {Array<string>} filesToLint
  */
-function initializeStream(globs, streamOptions) {
-  let stream = gulp.src(globs, streamOptions);
-  if (argv.watch) {
-    const watcher = lazypipe().pipe(watch, globs);
-    stream = stream.pipe(watcher()).pipe(debounce({wait: watchDebounceDelay}));
+async function runLinter(filesToLint) {
+  logLocalDev(green('Starting linter...'));
+  const eslint = new ESLint(options);
+  const results = {
+    errorCount: 0,
+    warningCount: 0,
+  };
+  const fixedFiles = {};
+  for (const file of filesToLint) {
+    const text = fs.readFileSync(file, 'utf-8');
```

Comment:
any reason why we use sync version?

Reply:
This is needed in order to live-stream the results for each file (in order). I'm trying to refactor this task this to run in parallel without live-streaming, at which point we won't need this.